### PR TITLE
[LGA-57] Build browser tests as a separate CI job

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

Adds `browser-test` job to CircleCI that (attempts) to run browser tests via Nightwatch.js and headless Chrome.

🚨 **Disabled most of the tests**
Most of the tests failed at the step `Scope diagnosis` with

```
 ✖   - URL is /scope/diagnosis  - expected "/scope/diagnosis" but got: "http://localhost:5000/session-expired"
```

All of these tests have been disabled, leaving only 2 running. The plan is to fix forward.

## Any other changes which would benefit highlighting?

**Version bumps**
I had to update the hardcoded versions of `selenium-server` and `chromedriver` to reflect the newer versions installed at the moment via `selenium-standalone`.

**Build image change**
The new job uses `circleci/python:2.7-node-browsers` as the builder image. To make the Python dependency cache reusable, I had to change the `test` job's image to `circleci/python:2.7`.

**Docker image built after both tests**
The workflow has slightly changed. Building the Docker image now requires both test jobs to complete successfully:

screen shot 2018-07-05 at 17 55 09

### CODE REVIEW

**Comment by rowlando** [comment]
>All of these tests have been disabled, leaving only 2 running. The plan is to fix forward.

@sldblog Have you raised a Jira issue to track this?

**Comment by sldblog** [comment]
@rowlando Yes, this morning: https://dsdmoj.atlassian.net/browse/LGA-77

File: .circleci/config.yml [hunk 1]
```diff
@@ -63,9 +60,68 @@ jobs:
...
+            ./node_modules/.bin/bower install
+            npm run update-selenium
+      - save_cache:
+          key: npm-v1-{{ checksum "package-lock.json" }}
+          paths:
+            - "./node_modules"
+
+      - run:
+          name: Compile assets
+          command: ./node_modules/.bin/gulp
```
Comment (rowlando) [inline .circleci/config.yml]: If this were an npm script, we could change this line to something like:

`npm run build`

I think it'd make it easier to understand how to build the assets from a new developer perspective.

**Comment by sldblog** [comment]
> is it worthwhile removing the references to PhantomJS and Firefox so it's clear.

@rowlando Done!

File: .circleci/config.yml (hunk 1 above)
Comment (rowlando) [reply .circleci/config.yml]: Ah, okay.

On reflection, I think the work to simplify the installation of front-end dependencies and building of assets could be done in separate piece of work - we want to remove bower anyway.

**Comment by sldblog** [comment]
🎉 🙇

**Pull request merged**